wgsl/index.bs

```diff
@@ -3959,6 +3959,278 @@ TODO: deduplicate these tables
+  <thead>
+    <tr><td>`<texture>` type                  <td>`offset`<td>`bias`  <td>`lod`   <td>`grad`
+  </thead>
+  <tr><td> `texture_sampled_1d<type>`         <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_sampled_1d_array<type>`   <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_sampled_2d<type>`         <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_2d_array<type>`   <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_3d<type>`         <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_cube<type>`       <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_cube_array<type>` <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_1d<type>`              <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_ro_1d_array<type>`        <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_ro_2d<type>`              <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_2d_array<type>`        <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_3d<type>`              <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_2d`                 <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_2d_array`           <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_cube`               <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_cube_array`         <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+</table>
+
+<table class='data'>
+  <caption>Compatibility with other named arguments</caption>
+  <thead>
+    <tr><td><td>`offset`<td>`bias` <td>`lod`  <td>`grad`
+  </thead>
+  <tr><td> `offset` <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `bias`   <td>&#10003;<td>        <td>        <td>
+  <tr><td> `lod`    <td>&#10003;<td>        <td>        <td>
+  <tr><td> `grad`   <td>&#10003;<td>        <td>        <td>
+</table>
+
+### Sample & Compare ### {#texture-sample-compare-functions}
+
+Sampling from a depth texture with a depth-comparison uses the `textureSampleCompare` function:
+
+```
+f32 textureSampleCompare(<depth_texture>, <sampler_comparison>, <coordinates>, f32 depth_reference [, offset: <offset>])
+```
+
+Where:
+
+* `<depth_texture>` is a [depth](#texture-depth) texture type.
+
+* `<sampler_comparison>` is a [sampler comparision](#sampler-type) type.
+
+* `<coordinates>` are the texture coordinates used for sampling.
+    The type for the texture coordinates depends on the `<texture>` type:
+
+    <table class='data'>
+      <thead>
+        <tr><td>`<texture>` type<td>`<coordinates>` type
+      </thead>
+      <tr><td> `texture_depth_2d`         <td> `vec2<f32>`
+      <tr><td> `texture_depth_2d_array`   <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube`       <td> `vec3<f32>`
```

Comment:
apparently, we can't even have depth cube textures in HLSL:
https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-samplecmp

Reply:
The page you linked suggests that you cannot use the optional `offset` for cube textures - I don't see it saying depth cubes are not supported?

Reply:
Yes, you are right, phew!